src/TestSuite/Fixture/TruncationStrategy.php

```diff
@@ -47,63 +46,28 @@ class TruncationStrategy implements StateResetStrategyInterface
     protected $tables = [];
 
     /**
-     * Constructor.
-     *
-     * @param bool $enableLogging Whether or not to enable query logging.
-     * @return void
+     * @var \Cake\TestSuite\Fixture\FixtureLoader
      */
-    public function __construct(bool $enableLogging = false)
-    {
-        $this->aliasConnections($enableLogging);
-    }
+    protected $fixtures;
 
     /**
-     * Alias non test connections to the test ones
-     * so that models reach the test database connections instead.
+     * Constructor.
      *
-     * @param bool $enableLogging Whether or not to enable query logging.
+     * @param \Cake\TestSuite\Fixture\FixtureLoader $fixtures The fixture loader being used.
      * @return void
      */
-    protected function aliasConnections(bool $enableLogging): void
+    public function __construct(FixtureLoader $fixtures)
     {
-        $connections = ConnectionManager::configured();
-        ConnectionManager::alias('test', 'default');
-        $map = [];
-        foreach ($connections as $connection) {
-            if ($connection === 'test' || $connection === 'default') {
-                continue;
-            }
-            if (isset($map[$connection])) {
-                continue;
-            }
-            if (strpos($connection, 'test_') === 0) {
-                $map[$connection] = substr($connection, 5);
-            } else {
-                $map['test_' . $connection] = $connection;
-            }
-        }
-        foreach ($map as $testConnection => $normal) {
-            ConnectionManager::alias($testConnection, $normal);
-            $connection = ConnectionManager::get($normal);
-            if ($connection instanceof Connection && $enableLogging) {
-                $connection->enableQueryLogging();
-            }
-        }
+        $this->fixtures = $fixtures;
```

Comment:
```suggestion
        $this->fixtures = $loader;
```

I don't know if `$this->fixtures` should be renamed or if used for other legacy code.

Reply:
There is no legacy here as this class is new. I can rename it.